nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -38,4 +63,30 @@ private static List<ApprovalStatus> generatePendingApprovalStatuses(List<URI> in
                                                       .withInstitutionId(uri)
                                                       .build()).toList();
     }
+
+    private boolean exists(URI publicationId) {
+        return nviCandidateRepository.findByPublicationId(publicationId).isPresent();
+    }
+
+    private void createCandidate(EvaluatedCandidateDto evaluatedCandidate) {
+        var candidateDetails = evaluatedCandidate.candidateDetailsDto();
+        var publicationDate = candidateDetails.publicationDateDto();
+        var pendingCandidate = new Candidate.Builder()
+                                   .withPublicationId(candidateDetails.publicationId())
+                                   .withIsApplicable(true)
+                                   .withCreators(mapToVerifiedCreators(candidateDetails.verifiedCreatorDtos()))
+                                   .withLevel(Level.parse(candidateDetails.level()))
+                                   .withInstanceType(candidateDetails.instanceType())
+                                   .withPublicationDate(mapToPublicationDate(publicationDate))
+                                   .withApprovalStatuses(
+                                       generatePendingApprovalStatuses(extractInstitutionIds(candidateDetails)))
+                                   .build();
+        nviCandidateRepository.save(pendingCandidate);
```

Comment:
extract to method, toPendingCandidate()

Reply:
creating toDao() method for every object will lead to something elegant like this:

`nviCandidateRepository.save(pendingCandidate.toDao());`